### PR TITLE
Data table

### PR DESCRIPTION
**Pull request by AMIRSOHAIL1998**
## Changes In PR
- [x] Added useDataTable hook and Docs.
- [x] Added Example for DataTable for render table with pagination. Added Docs for the Same.

### CODE REVIEW

**Review by rjvim — APPROVED** [review]
Is there a way to see "preview" of these changes? Or it has to be deployed?

**Comment by AMIRSOHAIL1998** [comment]
> Is there a way to see "preview" of these changes? Or it has to be deployed?

To preview We have to run it locally and check, otherwise we have to deploy it.

File: development/data-table-app/pages/data-table/index.mdx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+## How It Works
+The `DataTable` component uses the `useDataTable` hook from the @locoworks/reusejs-toolkit-react-hooks library to manage pagination and other functionalities. The hook returns various properties that are used to render the table and handle pagination. Please refer to the hook section to get more visiblity
+
+When you use the DataTable component, it automatically paginates the tableData based on the itemsPerPage prop. It also sets up navigation buttons to move between pages of data.
+## Example
+<Example />
+
+## Sample code demo
```
Comment (ShreyDhyani) [inline development/data-table-app/pages/data-table/index.mdx:51]: This is not needed. code preview is part of example only.

**Review by ShreyDhyani — CHANGES_REQUESTED** [review]

File: development/data-table-app/pages/data-table/index.mdx (hunk 1 above)
Comment (AMIRSOHAIL1998) [reply development/data-table-app/pages/data-table/index.mdx:51]: ok Removed

**Pull request merged**